docs/index-management/index-performance.md

```diff
@@ -466,13 +466,51 @@ increases. Often, such traversals can be rewritten as constrained
 traversals that can utilize a vertex-centric index to ensure acceptable
 performance at scale.
 
+### Using vertex-centric indexes on adjacent vertex ids
+
+In some cases it is relevant to find an edge based on properties of the adjacent vertex.
+Let's say we want to find out whether or not Hercules has battled Cerberus.
+```groovy
+h = g.V().has('name', 'hercules').next()
+g.V(h).out('battled').has('name', 'cerberus').hasNext()
+```
+
+A query like this can not use a vertex centric index because it filters
+on vertex properties rather than edge properties.
+But by restructuring the query, we can achieve exactly this.
+As both vertices are known, the vertex ids can be used to select the edge.
+```groovy
+h = g.V().has('name', 'hercules').next()
+c = g.V().has('name', 'cerberus').next()
+```
+
+In contrast to the name "Cebereus", which is a property of the adjacent vertex,
+the id of this vertex is already saved within the connecting edge itself.
+Therefore, this query runs much faster if hercules has battled many opponents:
+```groovy
+g.V(h).outE('battled').where(inV().is(c)).hasNext()
```

Comment:
@rngcntr Now I feel like this shouldn't be documented in the `vertex-centric index` section. In the above example, we are not using vertex-centric index at all. Maybe this should be moved to another place?

Reply:
Fair enough. Would it be sufficient to rename the section in line [469](https://github.com/JanusGraph/janusgraph/pull/1937/files/2d438977bc899d8293257bff839ab9d40fff4083#diff-97dc76a4f8809a5b2804ff398756d9cad89935a5d147824424234b02b85dcd0cR469)?

Reply:
I feel like it deserves to be in its own page, or maybe in the `JanusGraph Data Model` page? Now it is in the `index-performance` page which does not make much sense.

Reply:
The section is somehow related to indexes because it discusses the limits of vertex-centric indexes. However, I get your point that it should rather reside on a more specific page. A cool idea would be a separate page with dos and don'ts in Gremlin specific to JanusGraph's capabilities.
In case you have any other ideas for examples to mention there, feel free to compose a draft for such a page. Otherwise, we should open an issue to document the idea at least

Reply:
Sounds good!